walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -117,17 +108,34 @@ private boolean optionalAttendeesOnly(Event event, Collection<String> optionalAt
         Collection<TimeRange> possibleTimes = new ArrayList<TimeRange>();
         long requestedDuration = request.getDuration();
         boolean last = false;
+        
+        // sort events by starting time
+        Map<Integer, Event> eventStartTimes = new HashMap<>(); // key: event start time, value: event
+        for (Event event : events) {
+            int eventStart = event.getWhen().start();
+            eventStartTimes.put(eventStart, event);
+        }
+        List<Integer> eventStartTimesList = new ArrayList<>(eventStartTimes.keySet());
+        Collections.sort(eventStartTimesList);
+
+        List<Event> sortedEvents = new ArrayList<>();
+        for (int startTime : eventStartTimesList) {
+            sortedEvents.add(eventStartTimes.get(startTime));
+        }
```

Comment:
Let's read the documentation rather than reinventing the wheel. Java Collections API provides a `sort` function to do just this
https://docs.oracle.com/javase/7/docs/api/java/util/Collections.html#sort(java.util.List,%20java.util.Comparator)

https://stackoverflow.com/questions/16425127/how-to-use-collections-sort-in-java